cli/azd/extensions/microsoft.azd.extensions/internal/resources/languages/python/generated_proto/deployment_pb2_grpc.py

```diff
@@ -0,0 +1,143 @@
+# Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
+"""Client and server classes corresponding to protobuf-defined services."""
+import grpc
+import warnings
+
+import deployment_pb2 as deployment__pb2
+import models_pb2 as models__pb2
+
+GRPC_GENERATED_VERSION = '1.71.0'
+GRPC_VERSION = grpc.__version__
+_version_not_supported = False
+
+try:
+    from grpc._utilities import first_version_is_lower
+    _version_not_supported = first_version_is_lower(GRPC_VERSION, GRPC_GENERATED_VERSION)
+except ImportError:
+    _version_not_supported = True
+
+if _version_not_supported:
+    raise RuntimeError(
+        f'The grpc package installed is at version {GRPC_VERSION},'
+        + f' but the generated code in deployment_pb2_grpc.py depends on'
+        + f' grpcio>={GRPC_GENERATED_VERSION}.'
+        + f' Please upgrade your grpc module to grpcio>={GRPC_GENERATED_VERSION}'
+        + f' or downgrade your generated code using grpcio-tools<={GRPC_VERSION}.'
+    )
+
+
+class DeploymentServiceStub(object):
+    """Missing associated documentation comment in .proto file."""
+
+    def __init__(self, channel):
+        """Constructor.
+
+        Args:
+            channel: A grpc.Channel.
+        """
+        self.GetDeployment = channel.unary_unary(
+                '/azdext.DeploymentService/GetDeployment',
+                request_serializer=models__pb2.EmptyRequest.SerializeToString,
+                response_deserializer=deployment__pb2.GetDeploymentResponse.FromString,
+                _registered_method=True)
+        self.GetDeploymentContext = channel.unary_unary(
+                '/azdext.DeploymentService/GetDeploymentContext',
+                request_serializer=models__pb2.EmptyRequest.SerializeToString,
+                response_deserializer=deployment__pb2.GetDeploymentContextResponse.FromString,
+                _registered_method=True)
+
+
+class DeploymentServiceServicer(object):
+    """Missing associated documentation comment in .proto file."""
+
+    def GetDeployment(self, request, context):
+        """Gets the current environment.
+        """
+        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
+        context.set_details('Method not implemented!')
+        raise NotImplementedError('Method not implemented!')
+
+    def GetDeploymentContext(self, request, context):
+        """GetDeploymentContext retrieves the current deployment context.
+        """
+        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
+        context.set_details('Method not implemented!')
+        raise NotImplementedError('Method not implemented!')
```

Comment:
It looks like whatever GRPC generator you used is also outputting stub implementations of the server side component as well.  Is there a way to have it only generate the client components?

Reply:
@wbreza @Menghua1 
- should we move the proto files one level up for all languages to leverage? https://github.com/Azure/azure-dev/tree/main/cli/azd/extensions/microsoft.azd.extensions/internal/resources/languages/dotnet/proto
- why do we need to add the /generated_proto files? Shouldn't that be generated dynamically during building the extension project?

Reply:
I believe a lot of these tools need to generate the client libs from the proto files that are used by the code.